github/commitstatus.go

```diff
@@ -83,6 +87,7 @@ func (s CommitStatus) Add(sha, state, targetURL, description string) error {
 		return fmt.Errorf("JSON encode: %w", err)
 	}
 
+	s.log.Info("making a Github REST API http request")
```

Comment:
this is only to show the usage example, so that I can enable the feature and use it in the same commit....

It can be removed later or better say it will be replaced with a msg logging how many attempts we are doing...